app/plugin.go

```diff
@@ -450,3 +451,98 @@ func (a *App) notifyPluginEnabled(manifest *model.Manifest) error {
 
 	return nil
 }
+
+// GetPluginPublicKeys returns all public keys listed in the config.
+func (a *App) GetPluginPublicKeys() ([]*model.PublicKeyDescription, *model.AppError) {
```

Comment:
Do we need this method?

Reply:
Right now, not really. In future, may be. For me it's the issue of responsibility. app/plugin.go should be able to deliver the information about the plugin and the information flow should be consistent. I don't like getting information directly from config. This way will be better for unit tests as well(we will avoid mocking config).

Reply:
My main rational was to avoid additional dependency on config.

> model where we have to go through the domain api to get to domain specific config settings
Actually, this sounds good to me.